.github/workflows/test.yml

```diff
@@ -61,4 +61,4 @@ jobs:
           python --version
           python -m pytest --version
           firedrake-status
-          python -m pytest -v -n 6 --durations=40 --timeout=600 --cov=asQ --cov-report=term tests/
+          python -m pytest -v -n 6 --durations=40 --timeout=420 --cov=asQ --cov-report=term tests/
```

Comment:
what's the reason for this change?

Reply:
The longest test we have takes at maximum ~350s to run it its successful, and thats if the runners are very busy.
Reducing the timeout just means that if a test hangs then it fails faster.